subprojects/plugins/src/testFixtures/groovy/org/gradle/api/tasks/compile/AbstractCachedCompileIntegrationTest.groovy

```diff
@@ -37,7 +37,7 @@ abstract class AbstractCachedCompileIntegrationTest extends AbstractIntegrationS
         compileIsNotCached()
 
         when:
-        withBuildCache().succeeds 'clean', compilationTask
+        withBuildCache().succeeds 'clean', 'run'
```

Comment:
Why revert this?

Reply:
I introduced the change since I wanted to be able to re-use `AbstractCachedCompileIntegrationTest` for C++, too. I ended up not re-using the test for different reasons.
When using `run` as a task we do not only verify that we pulled from the cache but also that the actual result can be executed. That's why it is slightly better to use `run` and not only the `compilationTask` in this test.